census/estimator/trainer/model.py

```diff
@@ -0,0 +1,246 @@
+  """Build an estimator."""
+  
+  # Reused Transformations.
+  # Continuous columns can be converted to categorical via bucketization
+  age_buckets = layers.bucketized_column(
+      age, boundaries=[18, 25, 30, 35, 40, 45, 50, 55, 60, 65])
+
+  # Wide columns and deep columns.
+  wide_columns = [
+      # Interactions between different categorical features can also
+      # be added as new virtual features.
+      layers.crossed_column(
+          [education, occupation], hash_bucket_size=int(1e4)),
+      layers.crossed_column(
+          [age_buckets, race, occupation], hash_bucket_size=int(1e6)),
+      layers.crossed_column(
+          [native_country, occupation], hash_bucket_size=int(1e4)),
+      gender,
+      native_country,
+      education,
+      occupation,
+      workclass,
+      marital_status,
+      relationship,
+      age_buckets,
+  ]
+
+  deep_columns = [
+      layers.embedding_column(workclass, dimension=embedding_size),
+      layers.embedding_column(education, dimension=embedding_size),
+      layers.embedding_column(marital_status, dimension=embedding_size),
+      layers.embedding_column(gender, dimension=embedding_size),
+      layers.embedding_column(relationship, dimension=embedding_size),
+      layers.embedding_column(race, dimension=embedding_size),
+      layers.embedding_column(native_country, dimension=embedding_size),
+      layers.embedding_column(occupation, dimension=embedding_size),
+      age,
+      education_num,
+      capital_gain,
+      capital_loss,
+      hours_per_week,
+  ]
+
+  return tf.contrib.learn.DNNLinearCombinedClassifier(
+      model_dir=model_dir,
+      linear_feature_columns=wide_columns,
+      dnn_feature_columns=deep_columns,
+      dnn_hidden_units=hidden_units or [100, 70, 50, 25])
+
+
+def parse_label_column(label_string_tensor): 
+  """Parses a string tensor into the label tensor
+  Args:
+    label_string_tensor: Tensor of dtype string. Result of parsing the
+    CSV column specified by LABEL_COLUMN
+  Returns:
+    A Tensor of the same shape as label_string_tensor, should return
+    an int64 Tensor representing the label index for classification tasks, 
+    and a float32 Tensor representing the value for a regression task.
+  """
+  # Build a Hash Table inside the graph
+  table = tf.contrib.lookup.string_to_index_table_from_tensor(
+      tf.constant(LABELS))
+
+  # Use the hash table to convert string labels to ints
+  return table.lookup(label_string_tensor)
+
+
+# ************************************************************************
+# YOU NEED NOT MODIFY ANYTHING BELOW HERE TO ADAPT THIS MODEL TO YOUR DATA
+# ************************************************************************
+
+
+def column_to_dtype(column):
+  if isinstance(column, layers.feature_column._SparseColumn):
+    return tf.string
+  else:
+    return tf.float32
+
+
+def serving_input_fn():
+    feature_placeholders = {
+        column.name: tf.placeholder(column_to_dtype(column), [None])
```

Comment:
When are we switching to a single string placeholder?
Can you add a TODO in the meantime.